pox-locking/src/events.rs

```diff
@@ -281,6 +326,10 @@ fn create_event_info_data_code(
                         max-amount: {max_amount},
                         ;; equal to args[5]
                         auth-id: {auth_id},
+                        ;; Get end cycle ID
+                        end-cycle-id: (some (burn-height-to-reward-cycle new-unlock-ht)),
```

Comment:
In this case you don't need the `some` because you'll always be returning a value

Reply:
It's a bit nice to have type consistency here (e.g. `end-cycle-id` is always an `optional`), rather than a union type like `uint | Optional(Some(uint))`, but I don't feel strongly. The API code ingesting this data currently expects a uniform `Optional`.

Reply:
The optionals are a bit cumbersome. Not really a big deal though. However, the `none` is only needed for "delegation" case, which doesn't impact stacking/locking until the actual `delegate-stack-...` calls. So this could be simplified -- I kept the `some` wrapping for now.

Reply:
I'm good with consistency, let's keep the optional 👍